File: Vending Machine/source/src/VendingMachine.cpp
#include "VendingMachine.hpp"

void VendingMachine::work()
{
    Option = -1;
    mOption = -1;
    try
    {
        Option = MainMenu();
    }
    catch (const IncorrectOption& ex)
    {
        std::cout << "\nError: " << ex.what() << std::endl;
        return;
    }
    
    if (Option == 11) {
        try
        {
            mOption = MaintenanceMenu();
        }
        catch (const IncorrectMTOption& ex)
        {
            std::cout << "\nError: " << ex.what() << std::endl;
            return;
        }
        if (mOption == 0) {
            return;
        }
        else {
            SelectMOption(mOption);
            return;
        }
    }
    else{
        product = SelectBev(Option);
        try
        {
            if (PaymentProcessing(product))
            {
                if (Option <= 5) {
                    try
                    {
                        MilkAndSugar(Option);
                    }
                    catch (const InvalidOperator& ex)
                    {
                        std::cout << "\nError: " << ex.what() << std::endl;
                        return;
                    }
                }
                else {
                    try
                    {
                        SoySauce();
                    }
                    catch (const InvalidOperator& ex)
                    {
                        std::cout << "\nError: " << ex.what() << std::endl;
                        return;
                    }
                }
                PrepDisp(product);
            }
        }
        catch (const InvalidPayment& ex)
        {
            std::cout << "\nError: " << ex.what() << std::endl;
            return;
        }
    }
}

File: Vending Machine (Python)/source/VendingMachine.py
from Feeder import Feeder
from Display import Display
from Controller import Controller

class VendingMachine(Display,Controller,Feeder):
    def __init__(self):
            self.__option = -1
            self.__moption = -1
            self.__product = ()
            super().__init__()

    def work(self):
        self.__option = super().MainMenu(super().GetMenu())
        if self.__option == 11:
            self.__moption = super().MaintenanceMenu(super().GetPassword())
            if self.__moption == 0: 
                return
            else:
                super().SelectMOption(self.__moption)
        else:
            self.__product = super().SelectBev(self.__option)
            if super().PaymentProcessing(self.__product):
                if self.__option < 6:
                    super().MilkAndSugar(self.__option)
                else:
                    super().SoySauce()
                super().PrepDisp(self.__product)